spec/pg/dsn_parser_spec.rb

```diff
@@ -32,6 +32,21 @@
       expect(subject.parse(s)).to eq(:host => "local host")
     end
 
+    it "leading space before key" do
+      s = " host = 'local host'"
```

Comment:
May not be for this PR, but a hostname can't contain spaces, so I would expect this to fail

Reply:
Looking at it, we do no validation of port values either so I think validation DSN components should be done separately.

But, there's no reason the parser tests should be using invalid examples.  I'll see if I can change the relevant ones to test the same feature but using a password value.

Reply:
Ok, I've fixed the tests using spaces in host field to instead use the password field since that field can contain spaces.

Reply:
Thanks!